packages/samples/react/src/scenarios/appointment-form/AvailableAppointmentsForm.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState } from 'react';
+import { KolButton, KolForm, KolHeading, KolInputDate, KolInputRadio, KolSpin } from '@public-ui/react';
+import { FormValues } from './AppointmentForm';
+import { ErrorList } from './ErrorList';
+import { Field, FieldProps, useFormikContext } from 'formik';
+import { fetchAvailableTimes } from './appointmentService';
+import { Option } from '@public-ui/components/src';
+
+export function AvailableAppointmentsForm() {
+	const form = useFormikContext<FormValues>();
+
+	const [sectionSubmitted, setSectionSubmitted] = useState(false);
+	const [availableTimes, setAvailableTimes] = useState<Option<string>[] | null>(null);
+
+	useEffect(() => {
+		let ignoreResponse = false;
+		setAvailableTimes(null);
+
+		if (form.values.date) {
+			fetchAvailableTimes().then(
+				(times) => {
+					if (!ignoreResponse) {
+						setAvailableTimes(times);
+						void form.setFieldValue('time', times[0].value);
```

Comment:
Hier war ich praktisch gezwungen, einen Wert vorauszuwählen: KoliBri markiert per default den ersten radio button als `checked`, auch wenn der value mit keiner der Optionen übereinstimmt (z.B. leerer value). Das ist vermutlich ein Bug, der in KoliBri behoben werden sollte, weil so keine "leere Auswahl" von Radio-Buttons möglich ist. Getestet mit `''`, `undefined` und `null`.